suricata/package.yaml

```
@@ -1,33 +1,1228 @@
+---
 id: suricata
 name: Suricata
 author: Tenzir
 author_icon: https://github.com/tenzir.png
-package_icon: https://suricata.io/wp-content/uploads/2023/09/Logo-Suricata-vert-R.png
-
+package_icon: |
+  https://suricata.io/wp-content/uploads/2023/09/Logo-Suricata-vert-R.png
 description: |
   A package to simplify the onboarding of logs emitted from a Suricata
   instance.
-
 inputs:
   eve-path:
     name: EVE Path
     description: A path where suricata writes eve.json formatted logs
     type: string
     default: /var/lib/suricata/eve.sock
-
+  eve-address:
+    name: EVE Address
+    description: A network address that suricata forwards formatted logs to
+    type: string
+    default: tcp://0.0.0.0:0000
 pipelines:
-  import:
-    name: Suricata Import
-    description: Ingest Suricata events into the internal storage of the node.
+  publish-from-file:
+    name: Suricata Publish
+    description: Ingest Suricata events from a fileinto a subscription topic.
+    restart-on-error: 5m
+    disabled: false
+    definition: |
+      // tql2
+      legacy "from file {{ inputs.eve-path }} --follow read suricata"
+      publish "suricata"
+  publish-from-network:
+    name: Suricata Publish
+    description: |
+      Ingest Suricata events from a network source into a subscription topic.
+    restart-on-error: 5m
+    disabled: true
+    definition: |
+      // tql2
+      legacy "from {{ inputs.eve-address }} read suricata"
+      // TODO: set policy for "buffer" operator in TQL2?
+      buffer 16Mi
```

```suggestion
      buffer 16Mi, policy="drop"
```

I think that should work, please check.

---

Alright, all the new TQL2 pipeline iterations are running and delivering data. I would say it works.